src/gsy_e/models/strategy/energy_parameters/forward_profile.py

```diff
@@ -0,0 +1,185 @@
+import csv
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict
+
+import pendulum
+
+from gsy_framework.enums import AvailableMarketTypes
+from gsy_e.gsy_e_core.util import d3a_path
+
+
+# The Standard Solar Profile energy parameters in this module are not used for Intraday Markets
+ALLOWED_MARKET_TYPES = [
+    AvailableMarketTypes.DAY_FORWARD,
+    AvailableMarketTypes.WEEK_FORWARD,
+    AvailableMarketTypes.MONTH_FORWARD,
+    AvailableMarketTypes.YEAR_FORWARD]
+
+
+class StandardProfileException(Exception):
+    """Exception for the StandardProfile."""
+
+
+class StandardProfileParser:
+    """Class representing the Standard Solar Profile for forward products."""
+
+    FILENAME = Path(d3a_path) / "resources/standard_solar_profile.csv"
+    MONTHS = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    FIELDNAMES = ["INTERVAL"] + MONTHS
+    SLOT_LENGTH = pendulum.duration(minutes=15)
+    SLOTS_IN_ONE_HOUR = pendulum.duration(hours=1) / SLOT_LENGTH
+
+    @classmethod
+    def parse(cls) -> Dict:
+        """Parse the standard solar profile.
+
+        Returns: a dict where the keys are the numbers of the months and the values are dicts of
+            {time: value} items.
+        """
+        data = defaultdict(dict)
+        with open(cls.FILENAME, "r", encoding="utf-8") as inf:
+            reader = csv.DictReader(inf, fieldnames=cls.FIELDNAMES, delimiter=";")
+            next(reader)  # Skip header line
+            for row in reader:
+                # Only keep time, discarding the date information to make the profile more general
+                slot = pendulum.parse(row["INTERVAL"], exact=True)
+                # The SSP is expressed in power (kW) units, so we convert it to energy (kWh)
+                for month_idx, month in cls._enumerate_months():
+                    data[month_idx][slot] = float(row[month]) / cls.SLOTS_IN_ONE_HOUR
+
+        return dict(data)
+
+    @classmethod
+    def _enumerate_months(cls):
+        return enumerate(cls.MONTHS, start=1)
+
+
+class ForwardTradeProfileGenerator:
+    """
+    Class to generate a new profile that spreads the trade energy across multiple market slots.
+
+    The values of this profile are obtained by scaling the values of the Standard Solar Profile.
+    """
+
+    _STANDARD_SOLAR_PROFILE = StandardProfileParser.parse()
+
+    def __init__(self, peak_kWh: float):
+        self._scaler = ProfileScaler(self._STANDARD_SOLAR_PROFILE)
+        self._scaled_capacity_profile = self._scaler.scale_by_peak(peak_kWh=peak_kWh)
+
+    def generate_trade_profile(
+            self, energy_kWh, market_slot, product_type):
+        """Generate a profile based on the market slot and the product type."""
+        trade_profile = self._scaler.scale_by_peak(peak_kWh=energy_kWh)
```

Comment:
I'd suggest to assert `energy_kWh` is a positive float.

Reply:
Done in [e75be3b](https://github.com/gridsingularity/gsy-e/pull/1515/commits/e75be3b218ac0605d6fa4af5ee1bd0b2180795ed)!